equinox/nn/_embedding.py

```diff
@@ -89,3 +96,313 @@ def __call__(
                 "`eqx.nn.Embedding()(x)` should be called with a scalar index `x`. "
                 "Use `jax.vmap` if you would like to index with multiple values."
             )
+
+
+class RotaryPositionalEmbedding(Module, strict=True):
+    """
+    A rotary positional encoding module, as described in the paper
+    "RoFormer: Enhanced Transformer with Rotary Position Embedding". While
+    this module can be used in any context, it is particularly useful for
+    providing positional information to transformer models.
+
+    !!! example
+        The following example demonstrates how to use `RotaryPositionalEmbedding` in
```

Comment:
Nit: this needs a newline to render correctly in the docs:
```
!!! example

    The following ...
```
(Try running `pip install -r docs/requirements.txt`, and then `mkdocs serve -> Control-C -> mkdocs serve`, to see what the result looks like. Note that you do need two consecutive `mkdocs serve` commands, as the first one compuptes and caches some information needed for the second one.)